src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -406,6 +666,52 @@ CHIP_ERROR DefaultSceneTableImpl::RemoveSceneTableEntryAtPosition(FabricIndex fa
     return fabric.RemoveScene(mStorage, scene.mStorageId);
 }
 
+CHIP_ERROR DefaultSceneTableImpl::GetAllSceneIdsInGroup(FabricIndex fabric_index, GroupId group_id, Span<SceneId> & scene_list)
+{
+    FabricSceneData fabric(fabric_index, mMaxScenesPerFabric, mMaxScenesGlobal);
+    SceneTableData scene(fabric_index);
+
+    auto * iterator     = this->IterateSceneEntries(fabric_index);
+    SceneId * list      = scene_list.data();
+    uint8_t scene_count = 0;
+
+    while (iterator->Next(scene) && scene_count < scene_list.size())
+    {
+        if (scene.mStorageId.mGroupId == group_id)
+        {
+            list[scene_count] = scene.mStorageId.mSceneId;
+            scene_count++;
+        }
+    }
+    scene_list.reduce_size(scene_count);
+
+    // Check that the buffer was big enough for all scenes within this group
```

Comment:
This check looks wrong to me, or rather the while loop condition seems wrong.  Let's say the iterator initially points to 1 scene and scene_list.size() starts off as 0.  Then `iterator->Next()` in the loop condition will return true, the `scene_count < scene_list.size()` check will be false, we will not put any scenes in the list. But here `more_scenes_left` will also be false.

Please make sure this is tested.